internal/journal/voljournal.go

```diff
@@ -204,6 +204,25 @@ type ImageData struct {
 	ImageAttributes *ImageAttributes
 }
 
+// Connection represents a real or virtual connection to a ceph cluster
+// that can make changes to the journal(s).
+type Connection struct {
+	config *Config
+	// connection metadata
+	monitors string
+	cr       *util.Credentials
+}
+
+// Connect establishes a new connection to a ceph cluster for journal metadata.
+func (cj *Config) Connect(monitors string, cr *util.Credentials) (*Connection, error) {
```

Comment:
do we need to return an error here? it always returns nil

Reply:
It's future proofing as I have upcoming code changes that will be able to return a real error. If I don't add error-handling to all the call sites now I end up doing some of the work twice.